sos/plugins/__init__.py

```diff
@@ -234,8 +246,10 @@ def do_file_sub(self, srcpath, regexp, subst):
                             % (subst, regexp, path))
             if not path:
                 return 0
+            flags = re.IGNORECASE if ignore_case else 0
             readable = self.archive.open_file(path)
-            result, replacements = re.subn(regexp, subst, readable.read())
+            result, replacements = re.subn(regexp, subst,
+                                           readable.read(), flags=flags)
```

Comment:
Has this been tested in the case that `regexp` is a compiled `re` object? I'm assuming any flags passed to `re.subn()` are given priority over existing flags in the `re` but I have no idea if that's actually the case and this has potential for confusing behaviour (we would need to document the real behaviour if we were to go in this direction).

This is actually the expected way for plugins to use these interfaces when their regex needs are more complex than the simple pattern and `re.subn()` the function implements: a caller then has complete control over the `re` object used for the substitution. This would still require some explanation in the plugin comments or commit to justify the additional complexity but it is how the interfaces were designed to be used (we really don't want to carry a bunch of one-off kwargs that aren't generally needed).